### DISCUSSION TITLE
[expo-video] Extract media notifications into a separate `expo-media-session` module

### DISCUSSION THREAD

**Discussion by justblender** [expo/expo#31809] [Proposals and Ideas]
## Summary

This RFC proposes to move media notifications (better known as the "Now Playing" notifications) from the `expo-video` package into a new, separate Expo module: `expo-media-session`. 

## Motivation

The primary motivation is to provide a unified and generic interface for managing media notifications, which can be utilized by both the `expo-audio` and `expo-video` modules. This approach aims to avoid code duplication and ensure consistency across all supported platforms (iOS, Android, and web).

Currently, media notifications are only implemented for the `expo-video` package. However, `expo-audio` users should also expect to have similar functionality. Implementing and maintaining this separately in both packages would lead to code duplication and potential inconsistencies. 

## Why `expo-media-session`?

The primary reason for naming the package `expo-media-session`, rather than `expo-media-notifications`, is that it aligns with the browser API for managing media notifications, which is known as the [Media Session API](https://developer.mozilla.org/en-US/docs/Web/API/Media_Session_API).

**Comment by tsapeta** [comment]
Thanks for starting the conversation! cc @behenate @alanjhughes

**Comment by behenate** [comment]
I think it's a valid idea, although I think we could use this thread to also discuss other similar cases such as `AudioSession` which will be managed by both audio and video. We have a PR up with caching for expo-video, the iOS implementation is pretty complicated and it would be pretty cool if we didn't have to repeat it for `expo-audio`. Maybe a broader package such as `expo-media-core` would make more sense, especially for future-proofing.
@alanjhughes @tsapeta

**Comment by sebnun** [comment]
Hi, is there any update about this?  Expo 52 and ```expo-video``` are now out of beta, but I see that ```expo-audio``` still doesn't expose a way to control playback from the lock screen. 

Thanks for your work with expo.

**Comment by behenate** [comment]
@sebnun No significant updates for now, sorry! We are planning to plan this and make final decisions before the SDK 53 release

**Comment by TerryMooreII** [comment]
I am very excited for this functionality to be added to the `expo-audio`.  I currently use React native track player only because it supports these controls and this is the one feature that is keeping me from switching to `expo-audio`. Also, that project is very slow to fix issues which leads me to believe that its soon to be a dead project.  

I have searched this repo and I dont see any updates on this and was curious if this is still slated for the SDK 53 release?

**Reply by radko93** [reply]
there are PRs so seems to be coming in expo 54

**Comment by trentcowden** [comment]
Also hoping for updates on this :) Our app would benefit greatly from being able to use lock screen controls/media control with bluetooth devices!

**Comment by janwirth** [comment]
Has it landed?

**Comment by MikeKovarik** [comment]
+1

**Comment by noeljackson** [comment]
I really desperately need this. Is there any way that someone can give me some guidance on starting this?

**Reply by MikeKovarik** [reply]
I ended up abusing expo-video to play mp3s. Turns out you don't even need expo-audio after all. Though there are some issues with some events not being fired on iOS.

**Reply by noeljackson** [reply]
Thanks @MikeKovarik! Do you know what events don't get fired or have a link to issue?

**Comment by tomasggarcia** [comment]
Here I am also eagerly awaiting this update. A workaround using expo-audio also would be great.

**Comment by lautenschlager-dev** [comment]
I totally support this proposal. Having media notifications integrated into `expo-audio` would make it much more feature-complete for apps that need to handle both audio and video playback. For example, course or meditation apps. It’s an essential feature for a consistent user experience across native expo modules.

**Comment by jkaufman** [comment]
Completely agree with the need to centralize session management. While the proposal is focused on playback, we see issues with `expo-video` clobbering audio session settings required for recording.

**Comment by skusez** [comment]
react-native-track-player package doesn't work with RN 0.82+ when building for android + seems like it won't be maintained. Fully support this and patiently waiting for this to drop!

**Comment by IvoTsochev** [comment]
+1

**Comment by imraann0** [comment]
any update on this
